man/fi_av.3.md

```diff
@@ -499,6 +501,33 @@ that the output from the AV insertion call is unchanged.  The provider will
 return an fi_addr_t value that maps to each address, and that value must be
 used for all data transfer operations.
 
+# PEER GROUPS
+
+Peer groups provide a direct mapping to HPC and AI communicator constructs.
```

Comment:
Does peer group have some overlap with av set, which is also used to identify the peers in a collective communications... I am wondering whether it could finally replace av set.

Reply:
The concepts are similar, but I don't see them merging.  AV sets have a heavy weight setup costs.  They are there as part of setting up collectives, so there's some assumption that creating an AV set involves communication with some fabric management agent to allocate and reserved network resources.  Peer groups currently don't have any setup defined.  Apps simply set some bits in the fi_addr_t which conceptually map to a peer group.  Because these are just bits in the address, they really could be used for anything that the app wants.  The requirement on the provider is that receive buffers must have similar bits set.

If we add a peer group creation call, that moves them closer to an AV set.  But it depends if the create call is synchronous or asynchronous.  It also depends if the peers who are part of the peer group must be identified.  That's true for AV sets, but not peer groups.  If the provider is implementing collective APIs, it needs that information.